package/MDAnalysis/analysis/rms.py

```diff
@@ -310,6 +309,20 @@ def process_selection(select):
     select['reference'] = asiterable(select['reference'])
     return select
 
+def weight_type_check(weights,atoms,selection):
+
+    if (np.array(weights).ndim == 1) & (np.array(weights).dtype 
+                            in (np.dtype('float64'),np.dtype('int64'))):
```

Comment:
Do you really need the bitwise operator `&` here? More pythonic would be a pure `and`